apps/api/src/app/widgets/usecases/get-subscriber-preference/get-subscriber-preference.usecase.ts

```diff
@@ -0,0 +1,62 @@
+import { Injectable } from '@nestjs/common';
+import { SubscriberPreferenceRepository, NotificationTemplateRepository, NotificationTemplateEntity } from '@novu/dal';
+import { GetSubscriberPreferenceCommand } from './get-subscriber-preference.command';
+
+@Injectable()
+export class GetSubscriberPreference {
+  private notificationTemplateRepository = new NotificationTemplateRepository();
+
+  constructor(private subscriberPreferenceRepository: SubscriberPreferenceRepository) {}
+
+  async execute(command: GetSubscriberPreferenceCommand): Promise<any> {
+    const templateList = await this.notificationTemplateRepository.getActiveList(
+      command.organizationId,
+      command.environmentId,
+      true
+    );
+
+    const templatesIds = templateList.map((template) => template._id);
+
+    const subscriberPreferences = await this.subscriberPreferenceRepository.findAll(
+      command.environmentId,
+      command.subscriberId,
+      templatesIds
+    );
+
+    const preferences = templateList.map((template) => {
+      const currSubscriberPreference = subscriberPreferences.find(
+        (preference) => preference._templateId === template._id
+      );
+
+      if (currSubscriberPreference) {
+        return { template: template, preference: currSubscriberPreference };
+      }
+
+      /*
+       * check template default fallback
+       * add here the template(or fallback) is critical as well
+       */
+
+      return getNoSettingFallback(template);
+    });
+
+    /*
+     * check what param the client need
+     * and create response interface
+     * remove useless params on repo query
+     */
+
+    return preferences;
+  }
+}
+
+function getNoSettingFallback(template: NotificationTemplateEntity) {
```

Comment:
Would suggest creating an interface for this aswell so later we can specify the response DTO for the API aswell.

Reply:
Thanks for the heads up i already did on the next branch 🔥